src/app/clusters/testing/ClusterTester.h

```diff
@@ -90,38 +96,115 @@ class ClusterTester
+    {
+        CHIP_ERROR err = CHIP_NO_ERROR;
+        TLV::TLVWriter writer;
+        writer.Init(mTlvBuffer);
+
+        TLV::TLVReader reader;
+
+        SuccessOrExit(err = request.Encode(writer, TLV::AnonymousTag()));
+        SuccessOrExit(err = writer.Finalize());
+
+        reader.Init(mTlvBuffer, writer.GetLengthWritten());
+        SuccessOrExit(err = reader.Next(TLV::kTLVType_Structure, TLV::AnonymousTag()));
+
+        return mCluster.InvokeCommand(mRequest, reader, mHandler.get());
+
+    exit:
+        return std::make_optional(app::DataModel::ActionReturnStatus(err));
+    }
+
+    // Simplified overload to invoke a command with just the command ID and data.
+    // Builds the request automatically using the internal cluster's paths.
+    // Ideal for quick tests without manual request construction.
     template <typename T>
-    std::optional<chip::app::DataModel::ActionReturnStatus> InvokeCommand(chip::CommandId commandId, const T & data,
-                                                                          app::CommandHandler * handler)
+    [[nodiscard]] std::optional<chip::app::DataModel::ActionReturnStatus> Invoke(chip::CommandId commandId, const T & data,
+                                                                                 app::CommandHandler * handler = nullptr)
     {
         const auto & paths = mCluster.GetPaths();
         VerifyOrReturnError(paths.size() == 1u, CHIP_ERROR_INCORRECT_STATE);
         const chip::app::DataModel::InvokeRequest request = { .path = { paths[0].mEndpointId, paths[0].mClusterId, commandId } };
 
-        constexpr size_t kTlvBufferSize = 128; // Typically CommanderSender will use a TLV of size kMaxSecureSduLengthBytes. For
-                                               // now, just use 128 for the unit test.
-        uint8_t buffer[kTlvBufferSize];
         chip::TLV::TLVWriter tlvWriter;
-        tlvWriter.Init(buffer);
+        tlvWriter.Init(mTlvBuffer);
         ReturnErrorOnFailure(data.Encode(tlvWriter, chip::TLV::AnonymousTag()));
 
         chip::TLV::TLVReader tlvReader;
-        tlvReader.Init(buffer, tlvWriter.GetLengthWritten());
+        tlvReader.Init(mTlvBuffer, tlvWriter.GetLengthWritten());
         ReturnErrorOnFailure(tlvReader.Next());
 
-        return mCluster.InvokeCommand(request, tlvReader, handler);
+        // Use provided handler or internal one
+        app::CommandHandler * effectiveHandler = handler ? handler : mHandler.get();
+
+        return mCluster.InvokeCommand(request, tlvReader, effectiveHandler);
     }
 
+    chip::app::Testing::MockCommandHandler & GetHandler() { return *mHandler; }
+    const chip::app::Testing::MockCommandHandler & GetHandler() const { return *mHandler; }
+
+    template <typename ResponseType>
+    CHIP_ERROR DecodeResponse(ResponseType & response)
+    {
+        return mHandler->DecodeResponse(response);
+    }
+
+    const app::DataModel::InvokeRequest & GetRequest() const { return mRequest; }
```

Comment:
For all these we should explain WHY we expose them. In general I would prefer NOT to expose as much as possible (e.g. why do we have const versions for handler getters, why do we even have a handler getter, why do we access response, why do we split out a decoderesponse etc.).

please try to reduce the API surface.